src/Raven.Server/Documents/Transformers/CurrentTransformationScope.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using Raven.Server.Documents.Indexes.Static;
+
+namespace Raven.Server.Documents.Transformers
+{
+    public class CurrentTransformationScope
+    {
+        private readonly DocumentDatabase _documentDatabase;
+
+        public readonly IndexingFunc TransformResults;
+
+        [ThreadStatic]
+        public static CurrentTransformationScope Current;
```

Comment:
Note that we are using a lot of async calls, and may switch threads.
Maybe we should use AsyncLocal<T> instead? Alternatively, we need to make sure that we aren't doing any await when we do this.